src/HealthGPS/analysis_module.cpp

```diff
@@ -59,17 +59,32 @@ void AnalysisModule::initialise_vector(RuntimeContext &context) {
         factor_bin_widths_.push_back((max_factor - min_factor) / factor_bins_.back());
     }
 
-    // The number of factors to calculate is the number of factors minus the length of the `factors`
-    // vector.
-    size_t num_factors_to_calc = context.mapping().entries().size() - factors_to_calculate_.size();
+    // The number of factors to calculate stats for is the number of factors minus the length of the
+    // `factors` vector.
+    size_t num_stats_to_calc = context.mapping().entries().size() - factors_to_calculate_.size();
+
+    // But we calculate the mean and standard deviation of each factor, so we need to multiply by 2
+    num_stats_to_calc *= 2;
+
+    // We also want to calculate the prevalence and incidence of each disease
+    num_stats_to_calc += 2 * context.diseases().size();
+
+    // We also want to keep the count, deaths, and emigrations
+    num_stats_to_calc += 3;
+
+    // We also want to calculate the normal, overweight, obese, and above weight prevalence
+    num_stats_to_calc += 4;
+
+    // Finally, we want to calculate the mean and standard deviation of YLL, YLD, and DALY
+    num_stats_to_calc += 6;
```

Comment:
Yeah you're right it's a bit horrible isn't it. I'll have a think about a nicer way to do this. Eventually the other "channels" stuff will be removed and with it the reference to what the calculated stats (column headings) actually are, so it might be good to have a human-readable list somewhere perhaps stored as a `std::vector<std::string>`  as a data member in the class or something.